### PR TITLE
Enable sin, cos op for QNN HTP backend

### PR DESCRIPTION
**Pull request by Nick-Wei**
- add sin / cos to op_builder
- change quantizer to adopt new operator
- add test cases for both fp16 & quantized version
- fix bug in capture_program

### CODE REVIEW

**Comment by Nick-Wei** [comment]
Hi @haowhsu-quic, this PR enables `sin`, `cos` operator which are used in my own model and I also fix a bug related to `capture_program`. Hopefully we could have this merged, thanks.

File: backends/arm/third-party/ethos-u-core-driver
Comment (haowhsu-quic) [inline backends/arm/third-party/ethos-u-core-driver:1]: Please perform `git submodule update` and push again, thanks.

File: backends/qualcomm/utils/utils.py [hunk 1]
```diff
@@ -299,6 +299,7 @@ def get_decomp_table() -> Dict[torch._ops.OperatorBase, Callable]:
 def _transform(
     edge_program: ExportedProgram, custom_pass_config: FrozenSet[str] = frozenset()
 ) -> ExportedProgram:
+    custom_pass_config = custom_pass_config or {}
```
Comment (haowhsu-quic) [inline backends/qualcomm/utils/utils.py:302]: Since `custom_pass_config` is now default constructed, could you remove this line?

Comment (Nick-Wei) [reply backends/qualcomm/utils/utils.py]: It has already been removed, thank you.

**Review by haowhsu-quic — APPROVED** [review]
Looks good to me, thank you.

**Review by cccclai — APPROVED** [review]
Oops somehow I totally miss the PR, thank you for contributing!

**Pull request merged**